Remove Node Shutdown API feature flag

AthenaEryma opened the pull request:
This PR removes the Node Shutdown API feature flag.

The Node Shutdown API will now always be available.

Relates https://github.com/elastic/elasticsearch/issues/70338

AthenaEryma committed:
Remove Node Shutdown API feature flag

This PR removes the Node Shutdown API feature flag.

The Node Shutdown API will now always be available.

elasticmachine commented:
Pinging @elastic/es-core-infra (Team:Core/Infra)

elasticmachine commented:
Pinging @elastic/clients-team (Team:Clients)

sethmlarson reviewed (approved):
Approved for clients team

dakrone reviewed (approved):
LGTM, thanks Gordon

AthenaEryma committed:
Merge branch 'master' into decom/remove-feature-flag

AthenaEryma committed:
Check if xpack is enabled in cleanup

When I removed the feature flag, I assumed that we would always have the
Node Shutdown APIs, but that turns out not to be the case if xpack isn't
enabled. This case was caught by the logic to handle the case where the
feature flag wasn't enabled by accident.

This commit adds the check we always should have had.

AthenaEryma committed:
Merge branch 'master' into decom/remove-feature-flag

AthenaEryma committed:
Also check version before tyring cleanup

gwbrown commented:
Looks like the 7.15 build failed for the BWC checks.

@elasticmachine run elasticsearch-ci/bwc
@elasticmachine run elasticsearch-ci/part-2

AthenaEryma committed:
Merge branch 'master' into decom/remove-feature-flag

elasticsearchmachine commented:
## 💔 Backport failed
| Status | Branch | Result |
|:------:|:------:|:------:|
| ❌ |  7.15  | Commit could not be cherrypicked due to conflicts |
| ❌ |  7.x  | Commit could not be cherrypicked due to conflicts |

To backport manually run `backport --upstream elastic/elasticsearch --pr 76588`
